app/uk/gov/hmrc/gform/gform/NewFormController.scala

```diff
@@ -539,6 +543,14 @@ class NewFormController(
       }
     } yield res
 
+  private def hasItmpExpr(exprs: List[Expr]) =
+    exprs.contains(AuthCtx(AuthInfo.ItmpAddress)) ||
+      exprs.contains(AuthCtx(AuthInfo.ItmpName)) ||
+      exprs.contains(AuthCtx(AuthInfo.ItmpDateOfBirth)) ||
+      exprs.contains(AuthCtx(AuthInfo.ItmpNameLens(ItmpNameFocus.GivenName))) ||
+      exprs.contains(AuthCtx(AuthInfo.ItmpNameLens(ItmpNameFocus.MiddleName))) ||
+      exprs.contains(AuthCtx(AuthInfo.ItmpNameLens(ItmpNameFocus.FamilyName)))
```

Comment:
Problem with this is that it will fail to detect `AuthCtx(ItmpNameLens(ItmpNameFocus .GivenName))`  when part of bigger expression for example: 
```
${if 1 = 2 then auth.itmpName.givenName else 'NOPE'}
```
Can we fix that?